### PR TITLE
ExceptionHandling

### PR DESCRIPTION
**Pull request by grakeshrao**
1. Created GlobalExceptionHandler to handle the exceptions from all classes
2. Added Response codes for the custom exceptions.

### CODE REVIEW

File: src/main/java/org/c4sg/controller/GenericController.java [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
 
+@CrossOrigin
+@RestControllerAdvice
 public class GenericController {
 
-    @CrossOrigin
+	@CrossOrigin
```
Comment (felipealvesgnu) [inline src/main/java/org/c4sg/controller/GenericController.java:19]: Pay attention in the code identation. Fix them

File: src/main/java/org/c4sg/controller/ProjectController.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
 @RestController
 @RequestMapping("/api/projects")
 @Api(description = "Operations about Projects", tags = "project")
-public class ProjectController extends GenericController {
```
Comment (felipealvesgnu) [inline src/main/java/org/c4sg/controller/ProjectController.java:40]: Why did you delete the inheritance ? you can't

**Review by felipealvesgnu — CHANGES_REQUESTED** [review]

File: src/main/java/org/c4sg/controller/GenericController.java (hunk 1 above)
Comment (grakeshrao) [reply src/main/java/org/c4sg/controller/GenericController.java:19]: ok. I will fix the identation.

File: src/main/java/org/c4sg/controller/ProjectController.java (hunk 2 above)
Comment (grakeshrao) [reply src/main/java/org/c4sg/controller/ProjectController.java:40]: GenericController contains only individual exceptionHandlers. I configured the application to use GlobalExceptionHandler approach using @RestControllerAdvice. 
      Prior to my code change ProjectController was using GenericController exceptionHandler to throw the exception. With my new approach we don't need inheritance here
     Let me know if its other way around. Thanks

**Review by felipealvesgnu — APPROVED** [review]

**Comment by grakeshrao** [comment]
@felipealves-gnu. I was about to merge the identation changes and saw you fixed it. Thanks for the help.
